h2/src/main/org/h2/command/query/Select.java

```diff
@@ -150,7 +145,13 @@ public class Select extends Query {
     private boolean isForUpdate;
     private double cost;
     private boolean isQuickAggregateQuery, isDistinctQuery;
-    private boolean sortUsingIndex;
+
+    /**
+     * The length of the order-by prefix which is already sorted.
+     * A positive value triggers incremental sorting at the end.
+     * {@code Integer.MAX_VALUE} means that no final sort is needed.
+     */
+    private int alreadySorted;
```

Comment:
Maybe rename it to something like `numIndexSortedColumns`?

`alreadySorted` sounds like a boolean flag.